src/main/java/Parser.java

```diff
@@ -0,0 +1,135 @@
+public class Parser {
+    /**
+     * Parses the user input and processes the commands accordingly.
+     * Returns false to exit the program loop if the command is "bye",
+     * and returns true otherwise.
+     *
+     * @param input Input string entered by user.
+     * @param list Working task list of the program, for updating.
+     * @return false if user input is "bye", and true otherwise.
+     */
+    public boolean parse(String input, TaskList list) {
+        String[] commands = input.split(" ", 2);
+        switch (commands[0]) {
+        case "bye":
+            return false;
+        case "list":
+            System.out.println("    Here are the tasks in your list:");
+            int listLength = list.size();
+            for (int i = 0; i < listLength; i++) {
+                System.out.println("    " + (i + 1) + "." + list.get(i).getTaskDisplay());
+            }
+            break;
+        case "todo":
+            try {
+                Task todo = new ToDo(commands[1]);
+                list.add(todo);
+                printAdded(todo, list);
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("    Oops! The description of a todo cannot be empty.");
```

Comment:
the exception type does not correspond to the error message.
eg. index out of bound exception but print description cannot be empty.
and i dont think there is anything can trigger index out of bound exception here.